viewer/whylogs-script.js

```diff
@@ -0,0 +1,547 @@
+"use strict";
+
+(function () {
+  var MESSAGES = {
+    error: {
+      noInputElementFound: "It seems we could not find the file input element.",
+      noBrowserSupport: "This browser does not seem to support the `files` property of file inputs.",
+      noFileSelected: "Please select a file.",
+      fileAPINotSupported: "The file API is not supported on this browser yet.",
+      invalidJSONFile:
+        "It seems the JSON file you are trying to load is not valid due because it does not match whylogs format. Please check the file validity and try again.",
+    },
+  };
+
+  // HTML Elements
+  var $selectedProfile = $(".wl__selected-profile");
+  var $featureCount = $(".wl__feature-count");
+  var $sidebarFeatureNameList = $(".wl__sidebar-feature-name-list");
+  var $featureCountDiscrete = $(".wl__feature-count--discrete");
+  var $featureCountNonDiscrete = $(".wl__feature-count--non-discrete");
+  var $featureCountUnknown = $(".wl__feature-count--unknown");
+  var $tableBody = $(".wl__table-body");
+  var $featureSearch = $("#wl__feature-search");
+  var $featureFilterInput = $(".wl__feature-filter-input");
+  var $jsonForm = $("#json-form");
+  var $fileInput = $("#file-input");
+  var $tableContent = $("#table-content");
+  var $tableMessage = $("#table-message");
+  var $sidebarContent = $("#sidebar-content");
+  var $singleProfileWrap = $("#sidebar-content-single-profile");
+  var $multiProfileWrap = $("#sidebar-content-multi-profile");
+  var $profileDropdown = $("#sidebar-content-multi-profile-dropdown");
+  var $propertyPanelTitle = $(".wl-property-panel__title");
+  var $propertyPanelProfileName = $(".wl-property-panel__table-th-profile");
+
+  // Constants and variables
+  var batchArray = [];
+  var featureSearchValue = "";
+  var isActiveInferredType = {};
+  var propertyPanelData = [];
+  var profiles = [];
+  var jsonData = {};
+
+  // Util functions
+  function debounce(func, wait, immediate) {
```

Comment:
cant we use lodash? or trying to avoid dependencies?

Reply:
I can import only `debounce` from Lodash as it will be 2-3 functions only, nothing heavy and complex. However, I was trying to avoid external libraries if they are not necessary.

Reply:
I would leave it as is for now, because we are using only the debounce util function. Once we need one more function I will import Lodash. What do you think?